### PR TITLE
Add option to add custom icon to link

### PR DESCRIPTION
**Pull request by vineethasok**

### CODE REVIEW

File: src/components/Link/Link.tsx [hunk 1]
```diff
@@ -87,8 +89,8 @@ export const Link = <T extends ElementType = "a">({
       <IconWrapper $size={size}>
         <Icon
           name="popout"
-          className="external-icon"
-          data-testid="external-icon"
+          className={icon ?? "external-icon"}
```
Comment (serdec) [inline src/components/Link/Link.tsx:92]: i am not sure i understand how this should work. we pass the icon name as a class but the icon remains always "popout" ?

Comment (vineethasok) [reply src/components/Link/Link.tsx:92]: Sorry missed the place I should have added in the name too

Comment (vineethasok) [reply src/components/Link/Link.tsx]: Fixed it

File: src/components/Link/Link.tsx [hunk 2]
```diff
@@ -86,9 +88,9 @@ export const Link = <T extends ElementType = "a">({
     {hasIcon && (
       <IconWrapper $size={size}>
         <Icon
-          name="popout"
+          name={icon ?? "popout"}
```
Comment (serdec) [inline src/components/Link/Link.tsx:91]: storybook needs to be updated as well

Comment (vineethasok) [reply src/components/Link/Link.tsx:91]: https://click-ui-git-custom-icon-for-links-clickhouse.vercel.app/?path=%2Fdocs%2Ftypography-link--docs
Its there in storybook

**Review by serdec — APPROVED** [review]

**Pull request merged**